backend/routes/create_active_order.py

```
@@ -1,24 +1,97 @@
+
+    expiry = expiry.replace(
+        "Z", "+00:00"
+    )  # Replace Z with +00:00 to make it ISO 8601 compliant
+    expiry = datetime.fromisoformat(expiry)  # Convert expiry to datetime object
+    errors = []
+    if price < 1:
+        errors.append("Price must be greater than 0")
+    if quantity < 1:
+        errors.append("Quantity must be greater than 0")
+    if expiry < datetime.now(timezone.utc):
+        errors.append("Expiry date cannot be in the past")
+    if not supabase_middleman.fetch_stock_price(stock_id):
+        errors.append("Stock not found")
+    if errors:
+        return ", ".join(errors)
+
+    # validate user_id is a valid uuid and it exists
+    try:
+        UUID(hex=user_id)  # throws value error if not a valid uuid
+        user_profile = supabase_middleman.get_user_profile(user_id)
+        if not user_profile:
+            errors.append("User not found")
+    except ValueError:
+        errors.append("user_id is not in valid UUID format")
+
+    if errors:
+        return ", ".join(errors)
+
+    # validate that if order is a buy order, user has enough balance,
+    # or if order is a sell order, user has enough quantity
+    if buy_or_sell:  # Buy order
+        user_profile = supabase_middleman.get_user_profile(user_id)
+        if not user_profile:
+            errors.append("User not found")
+        if user_profile["balance"] < price * quantity:
+            errors.append("Insufficient balance")
+    else:  # Sell order
+        # portfolio = supabase_middleman.fetch_portfolio(user_id, stock_id)
+        portfolio = supabase_middleman.get_user_portfolio(user_id)[stock_id]
+        if not portfolio:
+            errors.append("Stock not found in users portfolio")
+        if portfolio["quantity"] < 1:
+            errors.append("User does not own stock")
+        if portfolio["quantity"] < quantity:
+            errors.append("Insufficient quantity")
+
+    if errors:
+        return ", ".join(errors)
```

I really think this should raise an HTTP exception to let the client know something went wrong (ie).

```py
        raise HTTPException(status.HTTP_400_BAD_REQUEST, errors)
```

---

currently had to ditch the http exceptions as the original author couldnt finish the pr and the http exceptions he implemented were throwing errors that I was unable to debug. so for now this matches the other api endpoints with string returns until we can do a sweeping http exception overhaul later.